### PR TITLE
Delete transport_interface.h and have unit tests use coreMQTT's version

### PR DESCRIPTION
**Pull request by sarenameas**
This PR is NOT ready. I need to update the demo CMakelists.txt

*Description of changes:*
coreMQTT is the Single Source of Truth for the transport_interface.h. 
This PR:
- updates the unit tests in platform/posix/transport to use the coreMQTT transport_interface.h.
- conditionally clone coreMQTT for the transport unit tests if it does not exist.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: integration-test/mqtt/CMakeLists.txt [hunk 1]
```diff
@@ -85,4 +85,4 @@ if(CLIENT_PRIVATE_KEY_PATH)
         ${stest_name} PRIVATE
         CLIENT_PRIVATE_KEY_PATH="${CLIENT_PRIVATE_KEY_PATH}"
     )
-endif()
```
Comment (sarenameas) [inline integration-test/mqtt/CMakeLists.txt:88]: Oops fixing

File: platform/posix/transport/CMakeLists.txt [hunk 2]
```diff
@@ -1,51 +1,66 @@
...
-                         PUBLIC 
+target_link_libraries( plaintext_posix
+                       PUBLIC 
                            sockets_posix )
 
 # Create target for POSIX implementation of OpenSSL.
 add_library( openssl_posix
                 ${OPENSSL_TRANSPORT_SOURCES} )
 
 set(OPENSSL_USE_STATIC_LIBS TRUE)
```
Comment (yourslab) [inline platform/posix/transport/CMakeLists.txt:40]: spacing of this one but it is not a big deal for me tbh

File: platform/posix/transport/CMakeLists.txt [hunk 3]
```diff
@@ -1,51 +1,66 @@
 # Include filepaths for source and include.
 include(${PLATFORM_DIR}/posix/posixFilePaths.cmake)
```
Comment (yourslab) [inline platform/posix/transport/CMakeLists.txt:2]: spacing here

**Review by yourslab — APPROVED** [review]

File: platform/posix/transport/CMakeLists.txt (hunk 2 above)
Comment (sarenameas) [reply platform/posix/transport/CMakeLists.txt:40]: Oh missed some thanks. yea the spacing in the file was bothering me lol

**Review by yourslab — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**